Configurations/15-android.conf

```diff
@@ -76,12 +56,18 @@
                 if (which("clang") !~ m|^$ndk/.*/prebuilt/([^/]+)/|) {
                     die "no NDK clang on \$PATH";
                 }
+                my $host=$1;
                 # harmonize with gcc default
-                (my $tridefault = $triarch) =~ s|^arm-|armv5te-|;
-                $cflags .= " -target $tridefault -gcc-toolchain "
-                        . "\$(ANDROID_NDK)/toolchains/$triarch-4.9/prebuilt/$1";
+                (my $tridefault = $triarch) =~ s/^arm-/armv5te-/;
```

Comment:
How about using armv7 for NDK >= r17 and armv5te otherwise? ARMv5 is deprecated in [NDK r16](https://github.com/android-ndk/ndk/wiki/Changelog-r16) and its support will be dropped since [NDK r17](https://github.com/android-ndk/ndk/wiki/Changelog-r17-beta1). I got linking errors about missing ```__atomic_*``` symbols in NDK r17 beta 1, which can be work-arounded by using armv7.

Reply:
As a technical detail. The reason for why armv7 code doesn't have `__atomic_*` references is because there are atomic instructions sequences that compiler can and does generate inline. For pre-armv7 cases atomic operations *might* require some assistance from kernel, which is why they are collected in library where proper course of action is determined.

As for suggestion to make it version-dependent. One would need to detect the version, and I'd be reluctant to use $ANDROID_NDK value. Because nothing says that it contains the version in its value. It be more appropriate to examine a file in $ANDROID_NDK...